esphome/components/tcs34725/tcs34725.cpp

```diff
@@ -149,14 +148,18 @@ void TCS34725Component::calculate_temperature_and_lux_(uint16_t r, uint16_t g, u
   g2 = g - ir;
   b2 = b - ir;
 
+  // discarding super low values? not recemmonded, and avoided by using auto gain.
   if (r2 == 0) {
-    return;
+    ESP_LOGW(TAG, "No light detected on red channel, switch to auto gain or adjust timing, values will be unreliable");
+    // legacy code
+    if (!integration_time_auto_){
```

Comment:
```suggestion
    if (!this->integration_time_auto_) {
```
```suggestion
    this->integration_time_auto_ = false;
```

See https://esphome.io/guides/contributing.html#codebase-standards

Reply:
fixed in https://github.com/esphome/esphome/pull/3258/commits/1972d5e78f768fc501c80b33c09e107b8739a577